src/utils/splitAtom.ts

```diff
@@ -71,9 +78,16 @@ export function splitAtom<Item, Key>(
         set: Setter,
         update: SetStateAction<Item>
       ) => {
-        const index = currentKeyList?.indexOf(key)
-        if (index === undefined || index === -1) {
-          throw new Error('index not found')
```

Comment:
Let's keep throwing error on write, until we see real use case issues.

Reply:
Done.